File: src/main/java/krasa/editorGroups/actions/TogglePanelVisibilityAction.java
package krasa.editorGroups.actions;

import com.intellij.openapi.actionSystem.ActionUpdateThread;
import com.intellij.openapi.actionSystem.AnActionEvent;
import com.intellij.openapi.project.DumbAwareAction;
import krasa.editorGroups.EditorGroupsSettingsState;
import krasa.editorGroups.PanelRefresher;
import org.jetbrains.annotations.NotNull;

import java.util.Objects;

public class TogglePanelVisibilityAction extends DumbAwareAction {
  @Override
  public @NotNull ActionUpdateThread getActionUpdateThread() {
    return ActionUpdateThread.BGT;
  }

  @Override
  public void actionPerformed(AnActionEvent e) {
    EditorGroupsSettingsState state = EditorGroupsSettingsState.state();
    state.setShowPanel(state.isShowPanel());
    PanelRefresher.getInstance(Objects.requireNonNull(getEventProject(e))).refresh();
  }


  @Override
  public void update(@NotNull AnActionEvent e) {
    super.update(e);
    EditorGroupsSettingsState state = EditorGroupsSettingsState.state();
    if (state.isShowPanel()) {
      e.getPresentation().setText("Hide Panel");
    } else {
      e.getPresentation().setText("Show Panel");
    }
  }
}

File: src/main/java/krasa/editorGroups/actions/TogglePanelVisibilityAction.kt
package krasa.editorGroups.actions

import com.intellij.openapi.actionSystem.ActionUpdateThread
import com.intellij.openapi.actionSystem.AnActionEvent
import com.intellij.openapi.project.DumbAwareAction
import krasa.editorGroups.messages.EditorGroupsBundle.message
import krasa.editorGroups.services.PanelRefresher
import krasa.editorGroups.settings.EditorGroupsSettings
import krasa.editorGroups.support.Notifications

class TogglePanelVisibilityAction : DumbAwareAction() {
  override fun getActionUpdateThread(): ActionUpdateThread = ActionUpdateThread.BGT

  override fun actionPerformed(e: AnActionEvent) {
    PanelRefresher.getInstance(getEventProject(e)!!).refresh()
    Notifications.notifyState(message("editor.groups.panel"), false)
  }

  override fun update(e: AnActionEvent) {
    super.update(e)

    when {
      EditorGroupsSettings.instance.isShowPanel -> e.presentation.text = message("action.hide.panel.text")
      else                                      -> e.presentation.text = message("action.show.panel.text")
    }
  }

  companion object {
    const val ID = "krasa.editorGroups.TogglePanelVisibility"
  }
}
